ext/memcached_rb/memcached_rb_behavior.c

```diff
@@ -1,143 +1,171 @@
+	"MEMCACHED_BEHAVIOR_USER_DATA",
+	"MEMCACHED_BEHAVIOR_SORT_HOSTS",
+	"MEMCACHED_BEHAVIOR_VERIFY_KEY",
+	"MEMCACHED_BEHAVIOR_CONNECT_TIMEOUT",
+	"MEMCACHED_BEHAVIOR_RETRY_TIMEOUT",
+	"MEMCACHED_BEHAVIOR_KETAMA_WEIGHTED",
+	"MEMCACHED_BEHAVIOR_KETAMA_HASH",
+	"MEMCACHED_BEHAVIOR_BINARY_PROTOCOL",
+	"MEMCACHED_BEHAVIOR_SND_TIMEOUT",
+	"MEMCACHED_BEHAVIOR_RCV_TIMEOUT",
+	"MEMCACHED_BEHAVIOR_SERVER_FAILURE_LIMIT",
+	"MEMCACHED_BEHAVIOR_IO_MSG_WATERMARK",
+	"MEMCACHED_BEHAVIOR_IO_BYTES_WATERMARK",
+	"MEMCACHED_BEHAVIOR_IO_KEY_PREFETCH",
+	"MEMCACHED_BEHAVIOR_HASH_WITH_PREFIX_KEY",
+	"MEMCACHED_BEHAVIOR_NOREPLY",
+	"MEMCACHED_BEHAVIOR_USE_UDP",
+	"MEMCACHED_BEHAVIOR_AUTO_EJECT_HOSTS",
+	"MEMCACHED_BEHAVIOR_NUMBER_OF_REPLICAS",
+	"MEMCACHED_BEHAVIOR_RANDOMIZE_REPLICA_READ",
+	"MEMCACHED_BEHAVIOR_CORK",
+	"MEMCACHED_BEHAVIOR_TCP_KEEPALIVE",
+	"MEMCACHED_BEHAVIOR_TCP_KEEPIDLE",
+	"MEMCACHED_BEHAVIOR_LOAD_FROM_FILE",
+	"MEMCACHED_BEHAVIOR_REMOVE_FAILED_SERVERS",
+	"MEMCACHED_BEHAVIOR_DEAD_TIMEOUT",
+	"MEMCACHED_BEHAVIOR_SERVER_TIMEOUT_LIMIT",
+	"MEMCACHED_BEHAVIOR_MAX"
 };
-#define MEMCACHED_BEHAVIORS_COUNT 39
+#define MEMCACHED_BEHAVIORS_COUNT (ARRAY_SIZE(MEMCACHED_BEHAVIOR_NAMES))
 
 const char* MEMCACHED_HASH_NAMES[] = {
-  "MEMCACHED_HASH_DEFAULT",
-  "MEMCACHED_HASH_MD5",
-  "MEMCACHED_HASH_CRC",
-  "MEMCACHED_HASH_FNV1_64",
-  "MEMCACHED_HASH_FNV1A_64",
-  "MEMCACHED_HASH_FNV1_32",
-  "MEMCACHED_HASH_FNV1A_32",
-  "MEMCACHED_HASH_HSIEH",
-  "MEMCACHED_HASH_MURMUR",
-  "MEMCACHED_HASH_JENKINS",
-  "MEMCACHED_HASH_MURMUR3",
-  "MEMCACHED_HASH_CUSTOM",
-  "MEMCACHED_HASH_MAX",
+	"MEMCACHED_HASH_DEFAULT",
+	"MEMCACHED_HASH_MD5",
+	"MEMCACHED_HASH_CRC",
+	"MEMCACHED_HASH_FNV1_64",
+	"MEMCACHED_HASH_FNV1A_64",
+	"MEMCACHED_HASH_FNV1_32",
+	"MEMCACHED_HASH_FNV1A_32",
+	"MEMCACHED_HASH_HSIEH",
+	"MEMCACHED_HASH_MURMUR",
+	"MEMCACHED_HASH_JENKINS",
+	"MEMCACHED_HASH_MURMUR3",
+	"MEMCACHED_HASH_CUSTOM",
+	"MEMCACHED_HASH_MAX",
 };
-#define MEMCACHED_HASH_COUNT 13
+#define MEMCACHED_HASH_COUNT (ARRAY_SIZE(MEMCACHED_HASH_NAMES))
 
 const char* MEMCACHED_DISTRIBUTION_NAMES[] = {
-  "MEMCACHED_DISTRIBUTION_MODULA",
-  "MEMCACHED_DISTRIBUTION_CONSISTENT",
-  "MEMCACHED_DISTRIBUTION_CONSISTENT_KETAMA",
-  "MEMCACHED_DISTRIBUTION_RANDOM",
-  "MEMCACHED_DISTRIBUTION_CONSISTENT_KETAMA_SPY",
-  "MEMCACHED_DISTRIBUTION_CONSISTENT_WEIGHTED",
-  "MEMCACHED_DISTRIBUTION_VIRTUAL_BUCKET",
-  "MEMCACHED_DISTRIBUTION_CONSISTENT_MAX"
+	"MEMCACHED_DISTRIBUTION_MODULA",
+	"MEMCACHED_DISTRIBUTION_CONSISTENT",
+	"MEMCACHED_DISTRIBUTION_CONSISTENT_KETAMA",
+	"MEMCACHED_DISTRIBUTION_RANDOM",
+	"MEMCACHED_DISTRIBUTION_CONSISTENT_KETAMA_SPY",
+	"MEMCACHED_DISTRIBUTION_CONSISTENT_WEIGHTED",
+	"MEMCACHED_DISTRIBUTION_VIRTUAL_BUCKET",
+	"MEMCACHED_DISTRIBUTION_CONSISTENT_MAX"
 };
-#define MEMCACHED_DISTRIBUTION_COUNT 8
+#define MEMCACHED_DISTRIBUTION_COUNT (ARRAY_SIZE(MEMCACHED_DISTRIBUTION_NAMES))
+
+static uint64_t lookup_behavior_const(memcached_behavior_t behavior, const char *cname)
```

Comment:
Thanks.. I am not sure about this API yet. but lets have it. I will add some tests.